compass_app/app/lib/common/config/dependencies.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:compass_app/features/results/business/usecases/search_destination_usecase.dart';
+import 'package:compass_app/features/results/data/destination_repository_local.dart';
+import 'package:compass_app/features/results/presentation/results_viewmodel.dart';
+import 'package:provider/provider.dart';
+import 'package:provider/single_child_widget.dart';
+
+/// Configure dependencies as a list of Providers
+List<SingleChildWidget> get providers {
```

Comment:
Is there an app architecture guarantee that this list will only be accessed when strictly necessary? I ask because I believe each access run the fill innards, reinitializing everything, instead of reusing anything.

Reply:
At the moment, this dependency tree is created on `runApp()` and indeed this function shouldn't be called elsewhere, so it is only called once to setup the top level Providers.

As the dependency tree becomes more complex (and we add config options to it) we will have to convert this getter into a function, and we should be able to cache all those dependencies that don't need to be rebuilt.

Another consideration could be making this a private method in the `main.dart` file to ensure it is not called elsewhere.

I think we will come with a better solution in the future when we tackle the Dependency Injection section in the Architecture docs.